### PR TITLE
TransformerTextField in the MC models

### PR DESCRIPTION
**Pull request by dirkgr**
This depends on https://github.com/allenai/allennlp/pull/5280 being merged first.

### CODE REVIEW

**Comment by dirkgr** [comment]
PIQA performs significantly worse in this branch. I'll have to find out why before merging.

File: allennlp_models/mc/models/transformer_mc.py [hunk 1]
```diff
@@ -89,6 +90,8 @@ def forward(  # type: ignore
         best_alternative : ``List[int]``
             The index of the highest scoring alternative for every instance in the batch
         """
+        self.eval()  # DEBUG
```
Comment (epwalsh) [inline allennlp_models/mc/models/transformer_mc.py:93]: Left over from debugging?

File: allennlp_models/mc/models/transformer_mc.py [hunk 2]
```diff
@@ -53,6 +53,7 @@ def __init__(
         )
 
         self._linear_layer = torch.nn.Linear(self._text_field_embedder.get_output_dim(), 1)
+        torch.manual_seed(1234)
```
Comment (epwalsh) [inline allennlp_models/mc/models/transformer_mc.py:56]: Left over from debugging?

File: allennlp_models/mc/models/transformer_mc.py (hunk 1 above)
Comment (dirkgr) [reply allennlp_models/mc/models/transformer_mc.py:93]: 😱

File: allennlp_models/mc/models/transformer_mc.py (hunk 2 above)
Comment (dirkgr) [reply allennlp_models/mc/models/transformer_mc.py:56]: 🙀

**Review by epwalsh — APPROVED** [review]
LGTM

**Pull request merged**